retrieval/scrape.go

```diff
@@ -151,14 +150,15 @@ func newScrapePool(ctx context.Context, cfg *config.ScrapeConfig, app Appendable
 	sp := &scrapePool{
 		appendable: app,
 		config:     cfg,
-		ctx:        ctx,
```

Comment:
Wait, this was never used/canceled? But it was the context passed to `newScrapeLoop` before which holds it and actually checks it in multiple places for cancelation. If that path was never possible, we'd have a logical flaw somehow, no?

Reply:
yes you are right it is referenced in the `mainLoop:` so this was never reachable so I added another commit
```
select {
		case <-sl.ctx.Done():
			close(sl.stopped)
			return
		case <-sl.scrapeCtx.Done():
			break mainLoop
		case <-ticker.C:
		}
```
I think the Scrape Loops logic needs a bit of refactoring as well, but  I will do that in another PR.